src/core/Akka/Actor/CoordinatedShutdown.cs

```diff
@@ -185,6 +185,19 @@ private UnknownReason()
             }
         }
 
+        /// <summary>
+        /// The shutdown was initiated by an ActorSystem termination hook
+        /// </summary>
+        public class ActorSystemTerminateReason : Reason
+        {
+            public static Reason Instance = new ActorSystemTerminateReason();
```

Comment:
Needs to be made `readonly`

Reply:
Looks like we have this issue with the other `Reason` implementations too. Could you also make all of those `readonly`?

Reply:
Sure